L1Trigger/L1TGlobal/src/GlobalBoard.cc

```diff
@@ -1015,11 +1015,22 @@ void l1t::GlobalBoard::runFDL(edm::Event& iEvent,
       if (bitValue) {
         // Make sure algo bit in range, warn otherwise
         if (iBit < prescaleFactorsAlgoTrig.size()) {
-          if (prescaleFactorsAlgoTrig.at(iBit) != 1) {
-            (m_prescaleCounterAlgoTrig.at(inBxInEvent).at(iBit))--;
-            if (m_prescaleCounterAlgoTrig.at(inBxInEvent).at(iBit) == 0) {
-              // bit already true in algoDecisionWord, just reset counter
-              m_prescaleCounterAlgoTrig.at(inBxInEvent).at(iBit) = prescaleFactorsAlgoTrig.at(iBit);
+          if (prescaleFactorsAlgoTrig.at(iBit) < 0) {
+            edm::LogWarning("L1TGlobal") << "Warning: Invalid negative prescale value "
+                                         << "(" << prescaleFactorsAlgoTrig.at(iBit) << ")\n"
+                                         << std::endl;
+
+            m_uGtAlgBlk.setAlgoDecisionInterm(iBit, false);
+          } else if (prescaleFactorsAlgoTrig.at(iBit) > 0 && prescaleFactorsAlgoTrig.at(iBit) < 1) {
+            edm::LogWarning("L1TGlobal") << "Warning: Invalid fractional prescale value between (0, 1) "
+                                         << "(" << prescaleFactorsAlgoTrig.at(iBit) << ")\n"
+                                         << std::endl;
+
+            m_uGtAlgBlk.setAlgoDecisionInterm(iBit, false);
+          } else if (prescaleFactorsAlgoTrig.at(iBit) != 1) {
+            const bool triggered = m_prescaleCounterAlgoTrig.at(inBxInEvent).at(iBit).accept();
+
```

Comment:
I would not do the check here, comparing doubles. I think it's better to do it in the ctor of `PrescaleCounter`.